<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SWinding</class>
 <widget class="QWidget" name="SWinding">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1158</width>
    <height>758</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>650</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="MPLCanvas2" name="w_viewer" native="true"/>
     </item>
     <item>
      <widget class="QLabel" name="in_wind_param">
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap resource="../../../Resources/pyleecan.qrc">:/images/images/MachineSetup/WindParam/Winding param.PNG</pixmap>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="widget" native="true">
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>250</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QGroupBox" name="g_pattern">
          <property name="title">
           <string>Pattern</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QComboBox" name="c_wind_type">
             <item>
              <property name="text">
               <string>Star of Slot</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>User Defined</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="in_Zs">
             <property name="text">
              <string>Slot number=123</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="in_p">
             <property name="text">
              <string>Pole pair number=32</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="in_qs">
               <property name="text">
                <string>Phases number</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="si_qs"/>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="in_Nlayer">
               <property name="text">
                <string>Layer number</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="si_Nlayer">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="in_coil_pitch">
               <property name="text">
                <string>Coil pitch</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="si_coil_pitch"/>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="in_Ntcoil">
               <property name="text">
                <string>Turns per coil</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QSpinBox" name="si_Ntcoil"/>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="in_Npcp">
               <property name="text">
                <string>Parallel circuits</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QSpinBox" name="si_Npcp">
               <property name="maximum">
                <number>999999999</number>
               </property>
               <property name="value">
                <number>12345</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="b_generate">
             <property name="text">
              <string>Generate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_import">
             <property name="text">
              <string>Import from CSV</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="g_edit">
          <property name="title">
           <string>Edit</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="in_Nslot">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Slot shift</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="si_Nslot">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="is_reverse">
             <property name="text">
              <string>Reversed winding</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_edit_wind_mat">
             <property name="text">
              <string>Edit Winding Matrix</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QGroupBox" name="g_output">
          <property name="minimumSize">
           <size>
            <width>200</width>
            <height>0</height>
           </size>
          </property>
          <property name="title">
           <string>Output</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="out_shape">
             <property name="minimumSize">
              <size>
               <width>175</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>Winding Matrix Shape : </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="out_ms">
             <property name="text">
              <string>ms = Zs / (2*p*qs) = ?</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="out_Nperw">
             <property name="toolTip">
              <string>Winding periodicity</string>
             </property>
             <property name="text">
              <string>Nperw</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="out_Ntspc">
             <property name="toolTip">
              <string>Winding number of turns in series per phase</string>
             </property>
             <property name="statusTip">
              <string>Winding number of turns in series per phase</string>
             </property>
             <property name="whatsThis">
              <string>Winding number of turns in series per phase</string>
             </property>
             <property name="text">
              <string>Ntspc: </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="out_Ncspc">
             <property name="toolTip">
              <string>Number of coils in series per parallel circuit</string>
             </property>
             <property name="statusTip">
              <string>Number of coils in series per parallel circuit</string>
             </property>
             <property name="whatsThis">
              <string>Number of coils in series per parallel circuit</string>
             </property>
             <property name="text">
              <string>Ncspc:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_export">
             <property name="text">
              <string>Export to CSV</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="b_preview">
       <property name="text">
        <string>Preview</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="b_previous">
       <property name="text">
        <string>Previous</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="b_next">
       <property name="text">
        <string>Next</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MPLCanvas2</class>
   <extends>QWidget</extends>
   <header>.....GUI.Tools.MPLCanvas</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../Resources/pyleecan.qrc"/>
  <include location="../../../Resources/pyleecan.qrc"/>
 </resources>
 <connections/>
</ui>
